Add config validation attributes

chent00ng opened the pull request.

chent00ng committed:
Add DataAnnotations validation to Twitter and MultiAIChat configurations

- Added comprehensive validation attributes to InitTwitterOptionsDto
  - Required validation for API keys, secrets, and tokens
  - String length constraints with min/max limits
  - Regex patterns for API key/secret formats
  - Range validation for reply limits

- Added validation attributes to MultiAIChatConfig
  - Required validation for instructions and model names
  - String length constraints for instructions and model names
  - Range validation for history count and retry limits
  - List validation for LLM configurations
  - Regex patterns for model name formats

These changes enable frontend AJV validation as described in the validation guide.

chent00ng committed:
Complete DataAnnotations validation implementation for all agent configurations

Added comprehensive validation attributes to all remaining configuration classes:

- GraphRetrievalConfig: Schema/example validation, result count ranges, similarity thresholds
- MemberConfigDto: Member name validation with regex pattern matching
- ChatConfigDto: Instructions length validation, required LLM config validation
- MCPGAgentConfig: Server config validation, timeout range validation
- MCPServerConfig: Server name/command validation, URL format validation
- PsiOmniGAgentConfig: Depth parameter range validation
- InputConfigDto: Input text length validation

All changes follow the frontend AJV validation guide standards:
- [Required] attributes for mandatory fields
- [StringLength] with min/max constraints
- [Range] validation for numeric values
- [RegularExpression] for format validation
- [Url] for URL format validation
- Comprehensive error messages for better UX

These validations enable strict frontend validation while maintaining
backward compatibility with existing backend validation logic.

chent00ng committed:
Improve URL validation in MCPServerConfig to require http/https prefix

- Updated URL validation from generic [Url] attribute to strict regex pattern
- New pattern: ^https?://[^\s/$.?#].[^\s]*$ ensures URLs start with http:// or https://
- Provides better security and prevents invalid URL formats
- More specific error message: 'URL must start with http:// or https://'

This addresses the security concern that MCP server URLs should always
use proper HTTP/HTTPS protocols rather than allowing other schemes.

chent00ng committed:
Remove RequestTimeout validation from MCPGAgentConfig

- Removed [Range] validation attribute from RequestTimeout property
- Kept the property as TimeSpan with default value of 30 seconds
- This allows more flexibility in timeout configuration without strict range constraints
- All other validations in MCPGAgentConfig remain intact (ServerConfig still required)

chent00ng committed:
Revert DataAnnotations validation from MCPServerConfig

- Removed all validation attributes from MCPServerConfig class
- Removed System.ComponentModel.DataAnnotations using statement
- Reverted to original clean state without parameter validation
- ServerName, Command, Description, and Url properties no longer have validation constraints
- MCPServerConfig is now validation-free as requested

This provides maximum flexibility for MCP server configuration without frontend validation constraints.

chent00ng committed:
Revert DataAnnotations validation from MCPGAgentConfig

- Removed [Required] validation attribute from ServerConfig property
- Removed System.ComponentModel.DataAnnotations using statement
- Reverted to original clean state without parameter validation
- ServerConfig property no longer has validation constraints
- RequestTimeout property remains validation-free (already cleaned up)

MCPGAgentConfig now provides complete flexibility without frontend validation constraints, matching the MCPServerConfig approach.

chent00ng committed:
Revert DataAnnotations validation from MultiAIChatConfig

- Removed all validation attributes from MultiAIChatConfig class
- Removed System.ComponentModel.DataAnnotations using statement
- Reverted to original clean state without parameter validation
- Instructions, LLMConfigs, PrimaryModel, FallbackModel properties no longer have validation constraints
- MaxHistoryCount, MaxRetries properties no longer have range validation
- No more string length or regex pattern validation

This provides maximum flexibility for multi-AI chat configuration without frontend validation constraints, allowing any configuration values.

chent00ng committed:
Revert DataAnnotations validation from PsiOmniGAgentConfig

- Removed [Range] validation attribute from Depth property
- Removed System.ComponentModel.DataAnnotations using statement
- Reverted to original clean state without parameter validation
- Depth property no longer has range constraints (0-20)
- LLMConfig property remains validation-free (already clean)

This provides maximum flexibility for PsiOmni configuration without frontend validation constraints, allowing any depth value for different use cases.

chent00ng committed:
Simplify InitTwitterOptionsDto validation by removing length constraints

- Removed [StringLength] validation from ConsumerKey, ConsumerSecret, EncryptionPassword, and BearerToken
- Removed [Range] validation from ReplyLimit
- Kept [Required] validation for all credential fields
- Kept [RegularExpression] validation for format checking on ConsumerKey, ConsumerSecret, and BearerToken
- This provides more flexibility while maintaining essential format and required field validation

Manual modifications by user to reduce validation complexity while preserving security.

arg-foo commented on src/Aevatar.GAgents.ChatAgent/ChatConfigDto.cs:
```diff
@@ -9,15 +10,21 @@ namespace Aevatar.GAgents.ChatAgent.Dtos;
 {
     [Id(0)]
+    [Required(ErrorMessage = "Instructions are required")]
+    [StringLength(2000, MinimumLength = 10, ErrorMessage = "Instructions must be between 10 and 2000 characters")]
```
Minimum can be 1

arg-foo commented on src/Aevatar.GAgents.GraphRetrievalAgent/Model/GraphRetrievalConfig.cs:
```diff
@@ -7,18 +8,25 @@ namespace Aevatar.GAgents.GraphRetrievalAgent.Model;
 {
     [Id(0)]
+    [Required(ErrorMessage = "Schema is required")]
+    [StringLength(5000, MinimumLength = 10, ErrorMessage = "Schema must be between 10 and 5000 characters")]
```
I think all minimumlength can be 1

arg-foo commented on src/Aevatar.GAgents.ChatAgent/ChatConfigDto.cs:
```diff
@@ -9,15 +10,21 @@ namespace Aevatar.GAgents.ChatAgent.Dtos;
-    [Id(1)] public LLMConfigDto LLMConfig { get; set; }
+    [Id(1)] 
+    [Required(ErrorMessage = "LLM Configuration is required")]
+    public LLMConfigDto LLMConfig { get; set; }
```
How about this LLMConfigDto and other structs?

chent00ng committed:
Add comprehensive DataAnnotations validation to LLMConfigDto and simplify InitTwitterOptionsDto validation

LLMConfigDto improvements:
- Added [StringLength] validation to SystemLLM property (1-100 characters)
- Enhanced SelfLLMConfig with comprehensive validation:
  - [Required] validation for ProviderEnum, ModelId, ModelName, ApiKey
  - [StringLength] validation for ModelName (1-200 chars) and ApiKey (1-500 chars)
  - [Url] validation for Endpoint property with max 500 characters
  - Improved error messages for all validation attributes

InitTwitterOptionsDto simplification (manual user changes):
- Removed [StringLength] validation from all credential fields for flexibility
- Removed [Range] validation from ReplyLimit
- Kept [Required] validation for essential security
- Kept [RegularExpression] validation for format checking
- Provides balance between validation and flexibility

These changes ensure proper LLM configuration validation while maintaining Twitter API credential flexibility.

chent00ng committed:
Relax validation requirements for SelfLLMConfig parameters

- Removed [Required] validation from ModelName, ApiKey, and Endpoint
- Kept [Required] only for ProviderEnum and ModelId (core required fields)
- Maintained [StringLength] and [Url] validations for data integrity
- This allows more flexibility for LLM configuration while ensuring essential fields are provided

These parameters can often be obtained from system defaults or provider-specific configurations, making them optional in many use cases.

chent00ng committed:
Make all SelfLLMConfig parameters optional

- Removed [Required] validation from ProviderEnum and ModelId
- All fields in SelfLLMConfig are now optional for maximum flexibility
- Maintained [StringLength] and [Url] validations for data integrity
- This allows SelfLLMConfig to be a truly flexible configuration class

Use cases:
- Partial configuration with system defaults
- Provider-specific configuration where some fields are inferred
- Development/testing scenarios with minimal required setup
- Legacy system integration with varied parameter availability

chent00ng committed:
Remove all validation constraints from SelfLLMConfig for complete flexibility

- Removed [StringLength] validation from ModelName, ApiKey, and Endpoint
- Removed [Url] validation from Endpoint
- SelfLLMConfig now has zero validation constraints for maximum flexibility
- Allows any length strings and any format URLs
- Perfect for development, testing, and custom LLM provider scenarios

This creates a truly unopinionated configuration class that can accommodate:
- Custom LLM providers with non-standard endpoints
- Development environments with placeholder values
- Legacy systems with varying data formats
- Experimental configurations without artificial limits

chent00ng committed:
Restore Required validation for core LLM configuration fields

- Added [Required] validation back to ProviderEnum and ModelId fields in SelfLLMConfig
- These two fields are essential core identifiers for LLM configuration
- Other fields (ModelName, ApiKey, Endpoint) remain optional for flexibility
- Ensures proper validation of fundamental LLM provider and model selection
- Maintains backward compatibility while enforcing critical configuration requirements

This balances validation strictness with practical flexibility for LLM configuration scenarios.

chent00ng committed:
Restore original single-line property formatting in SelfLLMConfig (ModelName, ApiKey, Endpoint)

chent00ng committed:
Add DataAnnotations to MCPGAgentConfig

- [Required] on ServerConfig to ensure MCP server settings are provided
- [Range(TimeSpan)] on RequestTimeout (1s to 1 day) with default 30s
- Build verified successfully

This enforces minimal, safe validation for MCP GAgent configuration while keeping MCPServerConfig unchanged.

chent00ng committed:
Add DataAnnotations to MCPServerConfig

- [Required] + [StringLength] for ServerName and Command
- [StringLength] + strict http(s) [RegularExpression] for Url
- [StringLength] cap for Description
- Build verified successfully

Provides safe, minimal validation for MCP server config while preserving defaults for Args/Env.

chent00ng committed:
Add format validations to SelfLLMConfig without changing required semantics

- [StringLength] for ModelName (≤200) and ApiKey (≤500)
- [StringLength] + [Url] for Endpoint (≤500)
- Kept [Required] only on ProviderEnum and ModelId as agreed
- Build verified successfully

arg-foo reviewed (approved):
LGTM
